include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -0,0 +1,111 @@
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
+
+#include <nbl/builtin/hlsl/subgroup/fft.hlsl>
+#include <nbl/builtin/hlsl/bit.hlsl>
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace workgroup
+{
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize>
+void store(NBL_REF_ARG(Accessor) sharedmemAccessor, uint32_t threadID, NBL_CONST_REF_ARG(complex_t<Scalar>) lo, NBL_CONST_REF_ARG(complex_t<Scalar>) hi) {
+    sharedmemAccessor.set(threadID + 0 * WorkgroupSize, lo.real());
+    sharedmemAccessor.set(threadID + 1 * WorkgroupSize, hi.real());
+    sharedmemAccessor.set(threadID + 2 * WorkgroupSize, lo.imag());  
+    sharedmemAccessor.set(threadID + 3 * WorkgroupSize, hi.imag());    
+}
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize>
+void load(NBL_CONST_REF_ARG(Accessor) sharedmemAccessor, uint32_t threadID, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) {
+    lo.real(sharedmemAccessor.get(threadID + 0 * WorkgroupSize));
+    hi.real(sharedmemAccessor.get(threadID + 1 * WorkgroupSize));
+    lo.imag(sharedmemAccessor.get(threadID + 2 * WorkgroupSize));  
+    hi.imag(sharedmemAccessor.get(threadID + 3 * WorkgroupSize));    
+}
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize, bool inverse>
+void FFT_loop(NBL_REF_ARG(Accessor) sharedmemAccessor, uint32_t step, uint32_t threadID){
+    const uint32_t sub_ix = threadID & (step - 1);
+    const uint32_t lo_x_ix = impl::bitfieldInsert(threadID, 0u ,sub_ix, 1u);
+    const uint32_t hi_x_ix = lo_x_ix | step;
+    const uint32_t lo_y_ix = lo_x_ix + 2 * WorkgroupSize;
+    const uint32_t hi_y_ix = hi_x_ix + 2 * WorkgroupSize;
+
+    sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+    complex_t<Scalar> lo = {sharedmemAccessor.get(lo_x_ix), sharedmemAccessor.get(lo_y_ix)};
+    complex_t<Scalar> hi = {sharedmemAccessor.get(hi_x_ix), sharedmemAccessor.get(hi_y_ix)};
+
+    fft::DIX<Scalar, inverse>::radix2(fft::twiddle<Scalar, inverse>(sub_ix, step << 1), lo, hi);
+
+    // Share results between threads
+    sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+    sharedmemAccessor.set(lo_x_ix, lo.real());
+    sharedmemAccessor.set(hi_x_ix, hi.real());
+    sharedmemAccessor.set(lo_y_ix, lo.imag());
+    sharedmemAccessor.set(hi_y_ix, hi.imag());     
+}
+
+// When doing a 1D FFT of size N, we call N/2 threads to do the work, where each thread of index 0 <= k < N/2 is in charge of computing two elements
+// in the output array, those of index k and k + N/2. The first is "lo" and the latter is "hi"
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize, uint32_t SubgroupSize, bool inverse>
+void FFT(NBL_REF_ARG(Accessor) sharedmemAccessor, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi){
+    const uint32_t doubleWorkgroupSize = WorkgroupSize << 1;
+    const uint32_t threadID = glsl::gl_LocalInvocationIndex();
```

Comment:
Neither the SPIR-V or Vulkan spec say that `LocalInvocationIndex` bijects with `(SubgroupID,SubgroupInvocationIndex)` in any specific way

Therefore you need to use (`workgroup::SubgroupContiguousIndex`)[https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/hlsl/workgroup/basic.hlsl#L26] instead of `glsl::gl_LocalInvocationIndex` to ensure the packing you rely on here.